sycl/plugins/opencl/CMakeLists.txt

```
@@ -9,9 +9,16 @@
 #TODO: remove dependency on pi.hpp in sycl project.
 #TODO: Currently, the pi.hpp header is common between sycl and plugin library sources. 
 #This can be changed by copying the pi.hpp file in the plugins project.
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../source/version.rc.in
+  ${CMAKE_CURRENT_BINARY_DIR}/version.rc
+  @ONLY)
+
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 add_library(pi_opencl SHARED
```

Would it make sense to change this to module?
```suggestion
add_library(pi_opencl MODULE
```

---

@smaslov-intel ,
Which one is right? We have used SHARED before. Now Alex suggests MODULE.

Here is the difference between SHARED and MODULE.

SHARED libraries are linked dynamically and loaded at runtime. MODULE libraries are plugins that are not linked into other targets but may be loaded dynamically at runtime using dlopen-like functionality.

---

MODULE sounds good, but what does it do differently technically?

---

I just saw @smaslov-intel comment in other part of this code. Since libpi_opencl.so is only dll-opened only, MODULE is the right choice.
I will accept Alex's change request.

---

> but what does it do differently technically?

Technically, MODULE makes our library libpi_opencl.so is not linkable dynamically. But, its content can be loaded dynamically via dllopen. So, I agree MODULE is the right choice in our case.

---

> MODULE makes our library libpi_opencl.so is not linkable dynamically

How is it possible?

---

@smaslov-intel if you use MODULE in target_link_libraries, CMake emits an error. No actual changes to binary, just build system semantics.

---

OK, sounds good to use MODULE.